pkg/parser/custom/py/setuppy.go

```diff
@@ -0,0 +1,101 @@
+package py
+
+import (
+	"strings" 
+	"github.com/safedep/vet/pkg/common/logger"
+	"github.com/google/osv-scanner/pkg/lockfile"
+	"github.com/safedep/vet/pkg/parser/custom/utils/regex_utils"
+)
+
+
+func ParseSetuppy(pathToLockfile string) ([]lockfile.PackageDetails, error) {
+	details := []lockfile.PackageDetails{}
+
+	// Get and print dependency strings from the setup.py file
+	if stringConstants, err := GetDependencies(pathToLockfile); err != nil {
+		return details, err
+	} else {
+		// fmt.Printf("Dependencies  %v", stringConstants)
+		for _, constant := range stringConstants {
+			pd := ParseRequirementsFileLine(constant)
+			details = append(details, pd)
+		}
+	}
+
+	logger.Debugf("Parsed Packaged Details %v", details)
+	return details, nil
+}
+
+// Return Dependency Strings in raw format such as "xxx>=123"
+func GetDependencies(pathToLockfile string) ([]string, error) {
+	setuppy_parser := NewSetuppyParserViaSyntaxTree()
+	// Get and print dependency strings from the setup.py file
+	if stringConstants, err := setuppy_parser.GetDependencyStrings(pathToLockfile); err != nil {
+		return nil, err
+	} else {
+		return stringConstants, nil
+	}
+}
+
+
+// todo: expand this to support more things, e.g.
+//
+//	https://pip.pypa.io/en/stable/reference/requirements-file-format/#example
+func ParseRequirementsFileLine(line string) lockfile.PackageDetails {
+	var constraint string
+	name := line
+
+	version := "0.0.0"
+
+	if strings.Contains(line, "==") {
```

Comment:
I think we are having some duplicate code here. Can we re-use the code from `pywheel` parser? May be we refactor all python related parsers into the `py` package and share the utils.

https://github.com/safedep/vet/blob/main/pkg/parser/pywheel.go#L84

Reply:
pywheel parser was not working for my case.